restapi.py

```diff
@@ -66,129 +103,138 @@ class Advance(Resource):
     def post(self):
         '''POST request with input data to advance the simulation one step
         and receive current measurements.'''
-        u = parser_advance.parse_args(strict=True)
-        y = case.advance(u)
-        return y
+        u = parser_advance.parse_args()
+        status, message, payload = case.advance(u)
+        return construct(status, message, payload)
+
 
 class Initialize(Resource):
     '''Interface to initialize the test case simulation.'''
 
     def put(self):
         '''PUT request to initialize the test.'''
-        args = parser_initialize.parse_args(strict=True)
-        start_time = float(args['start_time'])
-        warmup_period = float(args['warmup_period'])
-        y = case.initialize(start_time,warmup_period)
-        return y
+        args = parser_initialize.parse_args()
+        start_time = args['start_time']
+        warmup_period = args['warmup_period']
+        status, message, payload = case.initialize(start_time, warmup_period)
+        return construct(status, message, payload)
+
 
 class Step(Resource):
     '''Interface to test case simulation step size.'''
 
     def get(self):
         '''GET request to receive current simulation step in seconds.'''
-        step = case.get_step()
-        return step
+        status, message, payload = case.get_step()
+        return construct(status, message, payload)
 
     def put(self):
         '''PUT request to set simulation step in seconds.'''
-        args = parser_step.parse_args(strict=True)
+        args = parser_step.parse_args()
         step = args['step']
-        case.set_step(step)
-        return step, 201
+        status, message, payload = case.set_step(step)
+        return construct(status, message, payload)
+
 
 class Inputs(Resource):
     '''Interface to test case inputs.'''
 
     def get(self):
         '''GET request to receive list of available inputs.'''
-        u_list = case.get_inputs()
-        return u_list
+        status, message, payload = case.get_inputs()
+        return construct(status, message, payload)
+
 
 class Measurements(Resource):
     '''Interface to test case measurements.'''
 
     def get(self):
         '''GET request to receive list of available measurements.'''
-        y_list = case.get_measurements()
-        return y_list
+        status, message, payload = case.get_measurements()
+        return construct(status, message, payload)
+
 
 class Results(Resource):
     '''Interface to test case result data.'''
 
     def put(self):
-        '''PUT request to receive measurement data.'''
+        '''GET request to receive measurement data.'''
```

Comment:
Isn't `Results` a `PUT` request?

Reply:
Fixed in https://github.com/ibpsa/project1-boptest/pull/441/commits/9c334df9e01752ca7fe43166f049bbbdea5565dd.